### PR TITLE
Expose a interface to create all variables and enable the test of not creating variables every time.

### LINKED ISSUE #9653: Expose an interface to create all variables before executor.Run which specify create_vars to false
#9242 add an argument `create_vars` to `Executor.Run` to allow users to choose whether to create and destroy variables every time. If users want to specify `create_vars` to `false`, they also need to manually create all the needed variables before `Run`.

### PR DESCRIPTION
**Pull request by Xreki**
Fix #9653 

Sometimes users do not want to create and destroy variables in every run.

Here is the performance comparision for `test_inference_image_classificaion_resnet`.
GPU: Tesla K40m, CUDA 8.0, CUDNN v7.

|     | 1  | 2 | 4  | 8 | 16 | 32 | 64 | 128 | 256
|----|---|---|---|---|----|----|----|-----|-----|
|2018/03/14 | 866.048 | 852.3 | 858.021 | 987.86 | 1115.01 | 1556.01 | 2232.87 | 4375.75 | 8443.42|
|2018/03/31, not create variables in every run | 775.283 | 758.788 | 777.694 | 906.402 | 1045.67 | 1489.28 | 2155.31 | 4280.39 | 8330.89|
|speed up | 1.117073 | 1.123239 | 1.103289 | 1.08987 | 1.066312 | 1.044807 | 1.035986 | 1.022278 | 1.013508|

### CODE REVIEW

File: paddle/fluid/inference/tests/test_helper.h [hunk 1]
```diff
@@ -169,8 +169,14 @@ void TestInference(const std::string& dirname,
 
   // 6. Run the inference program
   {
+    const bool create_vars = false;
+    if (!create_vars) {
+      executor.CreateVariables(*inference_program, scope, 0);
```
Comment (Superjomn) [inline paddle/fluid/inference/tests/test_helper.h]: just some trivial issues, `!create_vars` ->  `executor.CreateVariables(` seems  a little wried.

Comment (Xreki) [reply paddle/fluid/inference/tests/test_helper.h]: `create_vars`意思是在`executor.Run`函数里面是否创建`Variables`，默认值为`true`，意思是每次`Run`都会create和destroy所有的`Variables`。如果`executor.Run`时设置参数`create_vars`为`false`，那么用户需要提前手动`CreateVaribales`，不然就会出现变量不存在的错误。

Comment (luotao1) [reply paddle/fluid/inference/tests/test_helper.h]: 能否增加一个单测说明`create_vars`为`false`时，用户如何提前手动`CreateVaribales`？

**Review by luotao1 — APPROVED** [review]
LGTM

File: paddle/fluid/inference/tests/test_helper.h (hunk 1 above)
Comment (Xreki) [reply paddle/fluid/inference/tests/test_helper.h]: Done。在image classification里面加了create_vars为false的单测。

**Pull request merged**